plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -378,6 +380,49 @@ public void testGetTableStatistics()
         this.jdbcClient.dropTable(SESSION, second);
     }
 
+    @Test
+    public void testCacheGetTableStatisticsWithQueryRelationHandle()
+    {
+        CachingJdbcClient cachingJdbcClient = cachingStatisticsAwareJdbcClient(FOREVER, true);
+        ConnectorSession session = createSession("some test session name");
+
+        JdbcTableHandle first = createTable(new SchemaTableName(schema, "first"));
+        JdbcTableHandle second = createTable(new SchemaTableName(schema, "second"));
+        JdbcTableHandle queryOnFirst = new JdbcTableHandle(
+                new JdbcQueryRelationHandle(new PreparedQuery("SELECT * FROM first", List.of())),
+                TupleDomain.all(),
+                Optional.empty(),
+                OptionalLong.empty(),
+                Optional.empty(),
+                0);
+
+        // load
```

Comment:
nit: it would be more readable to comment why zeroes are expected on the first commit. No hard push on that though, given this is intermediate commit.

Reply:
i think doesn't matter. i can squash commits if preferred though